packages/react/src/components/slot/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+
+export type Props<Name extends string> = {
+  children: React.ReactNode;
+  name: Name; // name of the slot the component is being rendered into
+  asChild?: boolean; // pass props to the immediate child
+};
+
+// TODO consider using asChild prop as an alternative to render props = passes props to the immediate child
+
+// add render props here
+
+/**
+ * TODO document this component
+ * - issue: it is not typesafe, it is possible to render a slot with a name that does not exist
+ * @returns
+ */
+export function Slot<N extends string>({ children, name, asChild }: Props<N>) {
+  if (asChild) {
```

Comment:
TODO out of scope for now IMO, I can add this in a follow up PR - don't want to create a large API immediately, got to be sure the design is sound.